gitops/utils/exceptions.py

```diff
@@ -10,7 +10,11 @@ class AppOperationAborted(Exception):
 class AppDoesNotExist(Exception):
     def __init__(self, app_name):
         # Funky hack to stomp on traceback.
-        self.args = warning(f"There's no app with the name '{app_name}', silly."),
+        if app_name:
+            message = f"There's no app with the name '{app_name}', silly."
+        else:
+            message = f"Could not find an 'apps' directory. Are you in a cluster repo?"
```

Comment:
Nice!

Reply:
> Do you also avoid running async if there's only a single server to run on?

Good point, it does now.

> Is it time to kill off the invoke commands in uptick-cluster?

Let's use async gitops in the wild for a few weeks so we can fall back on the invoke commands if we need to. Then I'd be confident removing them. I have a branch that removes them, I'll make a PR on uptick-cluster.